conf/modules.config

```
@@ -1,32 +1,410 @@
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
 ----------------------------------------------------------------------------------------
 */
+process {
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    ]
 
-params {
-    modules {
-        'fastqc' {
-            args = "--quiet"
-        }
-        'multiqc' {
-            args = ""
-        }
+    withName: SAMPLESHEET_CHECK {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: CAT_FASTQ {
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: FASTQC {
+        ext.args = '--quiet'
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: TRIMGALORE {
+        ext.args = [
+            '--fastqc',
+            params.trim_nextseq > 0 ? "--nextseq ${params.trim_nextseq}" : ''
+        ].join(' ').trim()
+        publishDir = [
+            [
+                path: { "${params.outdir}/trimgalore/fastqc" },
+                mode: 'copy',
+                pattern: "*.{html,zip}"
+            ],
+            [
+                path: { "${params.outdir}/trimgalore" },
+                mode: 'copy',
+                pattern: "*.txt"
+            ]
+        ]
+    }
+
+    withName: SEQTK_MERGEPE {
+        ext.prefix = { "${meta.id}.processed" }
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: BBMAP_BBNORM {
+        ext.args = [
+            params.bbnorm_target ? "target=${params.bbnorm_target}" : '',
+            params.bbnorm_min    ? "min=${params.bbnorm_min}" : '',
+        ].join(' ').trim()
+        publishDir = [
+            [
+                path : { "${params.outdir}/bbmap/bbnorm/logs" },
+                mode : 'copy',
+                pattern: "*.log"
+            ],
+            [
+                path : { "${params.outdir}/bbmap/bbnorm/"},
+                mode : 'copy',
+                pattern: "*.fastq.gz",
+                enabled: params.save_bbnorm_fastq
+            ]
+        ]
+    }
+
+    withName: BBMAP_BBDUK {
+        publishDir = [
+            pattern: '*.fastq.gz',
+            enabled: false
+        ]
+    }
+
+    withName: MEGAHIT_INTERLEAVED {
+        publishDir = [
+            path: { "${params.outdir}/megahit" },
+            mode: 'copy',
+            pattern: '**/*.{gz,log}'
+        ]
+    }
+
+    withName: WRITESPADESYAML {
+        publishDir = [
+            path: { "${params.outdir}/rnaspades" },
+            mode: 'copy',
```

should be done with all occurences I think:
```suggestion
            mode: params.publish_dir_mode,
```